File: src/main/java/dev/losterixx/sCore/features/spawn/SetSpawnCommand.java
package dev.losterixx.sCore.features.spawn;

import dev.dejvokep.boostedyaml.YamlDocument;
import dev.losterixx.sCore.Main;
import dev.losterixx.sCore.utils.ConfigManager;
import net.kyori.adventure.text.Component;
import net.kyori.adventure.text.minimessage.MiniMessage;
import org.bukkit.Location;
import org.bukkit.command.Command;
import org.bukkit.command.CommandExecutor;
import org.bukkit.command.CommandSender;
import org.bukkit.entity.Player;

public class SetSpawnCommand implements CommandExecutor {

    private MiniMessage mm = Main.mm;
    private Main main = Main.getInstance();
    private ConfigManager configManager = main.getConfigManager();
    private YamlDocument getConfig() { return configManager.getConfig("config"); }
    private YamlDocument getMessages() { return configManager.getConfig("messages"); }
    private YamlDocument getData() { return configManager.getConfig("data"); }
    private Component getPrefix() { return mm.deserialize(getConfig().getString("prefix")); }

    @Override
    public boolean onCommand(CommandSender sender, Command command, String label, String[] args) {

        if (!(sender instanceof Player)) {
            sender.sendMessage(getPrefix().append(mm.deserialize(getMessages().getString("general.noPlayer"))));
            return false;
        }

        if (!sender.hasPermission("sCore.command.setspawn")) {
            sender.sendMessage(getPrefix().append(mm.deserialize(getMessages().getString("general.noPerms"))));
            return false;
        }

        if (args.length != 0) {
            sender.sendMessage(getPrefix().append(mm.deserialize(getMessages().getString("commands.setspawn.usage"))));
            return false;
        }

        Player player = (Player) sender;
        Location loc = player.getLocation();

        getData().set("spawn.world", loc.getWorld().getName());
        getData().set("spawn.x", loc.getX());
        getData().set("spawn.y", loc.getY());
        getData().set("spawn.z", loc.getZ());
        getData().set("spawn.yaw", loc.getYaw());
        getData().set("spawn.pitch", loc.getPitch());

        configManager.saveConfig("data");

        player.sendMessage(getPrefix().append(mm.deserialize(getMessages().getString("commands.setspawn.set"))));

        return false;
    }

}

File: src/main/kotlin/dev/losterixx/sCore/features/spawn/SetSpawnCommand.kt
package dev.losterixx.sCore.features.spawn

import dev.losterixx.sCore.Main
import org.bukkit.command.Command
import org.bukkit.command.CommandExecutor
import org.bukkit.command.CommandSender
import org.bukkit.entity.Player

class SetSpawnCommand : CommandExecutor {

    private val mm = Main.miniMessage
    private val main = Main.instance
    private val configManager = Main.configManager
    private fun getConfig() = configManager.getConfig("config")
    private fun getMessages() = configManager.getConfig(getConfig().getString("langFile", "english"))
    private fun getData() = configManager.getConfig("data")
    private fun getPrefix() = getConfig().getString("prefix") ?: Main.DEFAULT_PREFIX

    override fun onCommand(sender: CommandSender, command: Command, label: String, args: Array<out String>): Boolean {

        if (sender !is Player) {
            sender.sendMessage(mm.deserialize(getPrefix() + getMessages().getString("general.noPlayer")))
            return false
        }

        if (!sender.hasPermission("sCore.command.setspawn")) {
            sender.sendMessage(mm.deserialize(getPrefix() + getMessages().getString("general.noPerms")))
            return false
        }

        if (!args.isEmpty()) {
            sender.sendMessage(mm.deserialize(getPrefix() + getMessages().getString("commands.setspawn.usage")))
            return false
        }

        val player = sender as Player
        val loc = player.location

        getData().set("spawn.world", loc.world?.name)
        getData().set("spawn.x", loc.x)
        getData().set("spawn.y", loc.y)
        getData().set("spawn.z", loc.z)
        getData().set("spawn.yaw", loc.yaw)
        getData().set("spawn.pitch", loc.pitch)
        configManager.saveConfig("data")

        sender.sendMessage(mm.deserialize(getPrefix() + getMessages().getString("commands.setspawn.set")))

        return false
    }

}